### ISSUE TITLE
Completely changes syntax highlighting of all YAML files

### ISSUE THREAD

**Issue by tristan957** [VSChina/vscode-ansible#93]
## Environment
- Extension version: 0.1.4
- OS Type: Windows/Linux/MacOS: MacOS

## Summary

This extension takes over all my YAML files. I don't need Ansible autocomplete on a YAML file not related to Ansible. It also completely changes the syntax highlighting from a normal YAML file.

## Reproduce steps

Install extension. Open YAML file

## Expected Results

Not to change the behavior of basic YAML files

## Actual Results

Behavior of basic YAML files was changed

**Comment by yaweiw** [comment]
@yungez

**Comment by yungezz** [comment]
published in v0.1.5. pls have a try and let us know if any issue. thanks again.

** Instruction to enable/disable ansible auto-completion **
1. Disable it completely by set `ansible.autocompletion` to `false`.
2. Enable it for only specific yaml files:
    - set `ansible.autocompletion` to 'false'
    - before first line of your ansible yml file, add this line `# ansible-configured`.

**Comment by tristan957** [comment]
Thanks guys. Fairly quick fix!

**Comment by tristan957** [comment]
The syntax highlighting is still being overwritten for all YAML files.

**Comment by tristan957** [comment]
I would like to point out that a month and half later, the issue still persists on version 0.2.6. It changes syntax highlighting of every YAML file...

**Comment by yungezz** [comment]
@tristan957  sure let's reopen the issue. Syntax highlighting is applied to all yaml file so far by design because of ansible playbook doesn't has its own syntax. We can add a configuration like the one for `autocompletion` to let user enable/disable the syntax highlighting. while for enabling it partially for selected yaml file it need more consideration. will update here later.

**Comment by jnunderwood** [comment]
The extension "language-Ansible", at https://marketplace.visualstudio.com/items?itemName=haaaad.ansible, allows for setting file associations. I think this would be a better approach than having separate settings for auto complete and syntax highlighting. We may have both ansible yaml files and other yaml files in the same project.

```
    "files.associations": {
        "**/defaults/**/*": "ansible",
        "**/tasks/**/*.yml" : "ansible",
        "**/handler/*.yml" : "ansible",
        "**/*_vars/**/*.yml" : "ansible",
        "**/roles/**/*.yml" : "ansible",
        "**/playbooks/**/*.yml" : "ansible",
        "**/*ansible*/**/*.yml" : "ansible",
        "**/vars/**/*.yml": "ansible",
        "**/inventory/*/*": "ansible"
    }
```

**Comment by jouve** [comment]
Right now, I have to enable the extension and reload vscode when I want to work with ansible and disable/reload again when I'm done.

**Comment by yungezz** [comment]
do configuration for syntax highlighting by `files.associations` to `ansible`. released in v0.3.0. pls try it and let us know if any feedback!

**Comment by tristan957** [comment]
I would like to thank the team for listening to feedback regarding this issue even after it had been originally closed.
